edge-modules/iotedge-diagnostics-dotnet/src/Program.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Diagnostics
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Sockets;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Concurrency;
+    using Microsoft.Azure.Devices.Edge.Util.TransientFaultHandling;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+    using ProxyLib.Proxy;
+    using ProxyLib.Proxy.Exceptions;
+
+    class Program
+    {
+        public static int Main(string[] args)
+        {
+            try
+            {
+                MainAsync(args).Wait();
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
+                return 1;
+            }
+
+            return 0;
+        }
+
+        static async Task MainAsync(string[] args)
+        {
+            var config = new ConfigurationBuilder().AddCommandLine(args).Build();
+            switch (args[0])
+            {
+                case "edge-agent":
+                    await EdgeAgent(config["management-uri"]);
+                    break;
+                case "iothub":
+                    await Iothub(config["hostname"], config["port"], config["proxy"]);
+                    break;
+                case "local-time":
+                    Console.WriteLine(DateTime.Now.ToUnixTimestamp());
+                    break;
+                default:
+                    throw new Exception("Invalid args");
+            }
+        }
+
+        static async Task EdgeAgent(string managementUri)
```

Comment:
Normally starts with verb for method name in C# and uses Async suffix for async method.